app/src/main/java/com/example/weathertimeapp/mvp/model/WeatherTimeModel.kt

```diff
@@ -12,26 +13,25 @@ import com.example.weathertimeapp.utils.NAME
 import io.reactivex.Observable
 import org.json.JSONArray
 
-class WeatherTimeModel(private val citiesJson: String) : WeatherTimeContracts.Model {
+class WeatherTimeModel(private val assetManager: AssetManager) : WeatherTimeContracts.Model {
 
     private val api: ForecastRequestGenerator = ForecastRequestGenerator()
 
-    override fun createListOfCities(listOfCities: ArrayList<City>): MutableList<String> {
-        val jsonArray = JSONArray(citiesJson)
-        val stringList = mutableListOf<String>()
-        for (i in 0 until jsonArray.length()) {
-            val jsonObject = jsonArray.getJSONObject(i)
-            if (jsonObject.get(COUNTRY) == COUNTRY_AR) {
+    override fun createListOfCities(): MutableList<String> {
+        val citiesArray = JSONArray(assetManager.open(FILE_NAME).bufferedReader().use { it.readText() })
+        val listOfCities = mutableListOf<String>()
+        for (i in 0 until citiesArray.length()) {
```

Comment:
Pls use a for-each to range json

Reply:
I search for a for-each to range it, but JSONArray doesn't have that function, the way i found to solved it is the one i used, can you show me the way you want to do it?

Reply:
i try to find a way to do a for-each but the JSONArray doesn't have and iterator function implemented, and the one i used it is the best solution i found. 
Can you show me the way you wanted to solved it?